README.md
## mKaskus Spoiler Enabler

Development version of **mKaskus Spoiler Enabler** on [openuserjs](https://openuserjs.org/scripts/zackad/mKaskus_Spoiler_Enabler). This is a userscript for [mobile kaskus](http://m.kaskus.co.id) to enabling spoiler feature that disabled on mobile/wap version. But you can only use it with desktop browser.

Developed and tested with Coolnovo browser (based on Chromium 27), other browser may or may NOT work. For simplicity, jQuery is used instead of vanilla javascript so maybe there's compatibility problem with some web browser.

## Feature

- show/hide spoiler by click
- hide origin grey link on post entry
- toggle spoiler with Shift+X hotkey
- last page link added on in myforum page
- open whoposted link on new tab
- redirect link removed
- convert kaskus link to wap version
- **New** : Load image link by click
- **New** : Show image size before load
- **New** : Show decimal point on image size

> **bold**: active on image off mode

CHANGELOG
v0.3.6.4
- convert kaskus link to wap version (m.kaskus)

v0.3.6.3
- redirect link removed [thanks : AMZZZMA]

v0.3.6.2
- last page link added in myforum page

v0.3.6.1
- show whoposted on new tab instead popup mode

v0.3.6
- decimal point added on image size

v0.3.5
- load image by click
- show image size before loading

v0.3.4
- activities link edited

v0.3.3
- Shift+X hotkey added [thanks : indrayoo]
- minor style fix

v0.3.2
- fixed bug (pict not showed on FF)

v0.3.1
- merged with kaskus-grey-link-remover

v0.3
- hide/show all spoiler button added

v0.2.2
- style fixed

v0.2.1
- style changed [Thanks : booster.bs]

v0.2
- grey link remover added (m.kaskus only)
- license change

v0.1
- init

mKse.v0.3.user.js
// ==UserScript==
// @name           mKaskus Spoiler Enabler
// @namespace      https://openuserjs.org/users/zackad/
// @homepageURL    http://www.kaskus.co.id/profile/4125324
// @description    Spoiler di m.kaskus layaknya versi desktop
// @author         zackad
// @version        0.3
// @include        http://m.kaskus.co.id/post/*
// @include        http://m.kaskus.co.id/thread/*
// @include        http://m.kaskus.co.id/lastpost/*
// @license        MIT License
// @grant          GM_addStyle
// @require        http://code.jquery.com/jquery-1.10.1.min.js
// @run-at         document-end
// ==/UserScript==
$(document).ready(function(){
    $('span[style*="font-size:10px;color: #888;"]').hide();
    
    var sContent = $('div[class*="content_spoiler"]');
    sContent.hide();
    var show_all = '<a class="show-all" class="btn bnt-sm" style="margin: 0 10px;background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Show All Spoiler">Show all spoiler</a>';
    var hide_all = '<a class="hide-all" class="btn bnt-sm" style="margin: 0 10px;background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide All Spoiler">Hide all spoiler</a>';
    var show_spoiler = '&nbsp&nbsp<a class="show-spoiler" class="btn btn-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px; border-radius:3px;" href="javascript:void(0);" title="Show Spoiler">Show</a>';
    var hide_spoiler = '<a class="hide-spoiler" class="btn bnt-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide Spoiler">Hide</a>';
    
    $('#bbcode_div > i').after(show_spoiler, hide_spoiler, show_all, hide_all);
    $('.hide-spoiler').hide();
    $('.hide-all').hide();
    
    $('.show-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).next().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.show();
    });
    
    $('.hide-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).prev().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.hide();
    });
    
    $('.show-all').click(function(){
        sContent.show();
        $('.show-all').hide();
        $('.show-spoiler').hide();
        $('.hide-spoiler').show();
        $('.hide-all').show();
    });
    
    $('.hide-all').click(function(){
        sContent.hide();
        $('.hide-all').hide();
        $('.hide-spoiler').hide();
        $('.show-spoiler').show();
        $('.show-all').show();
    });
});


mKse.v0.2.2.user.js
// ==UserScript==
// @name           mKaskus Spoiler Enabler
// @namespace      https://openuserjs.org/users/zackad/
// @homepageURL    http://www.kaskus.co.id/profile/4125324
// @description    Spoiler di m.kaskus layaknya versi desktop
// @author         zackad
// @version        0.2.2
// @include        http://m.kaskus.co.id/post/*
// @include        http://m.kaskus.co.id/thread/*
// @include        http://m.kaskus.co.id/lastpost/*
// @license        MIT License
// @grant          GM_addStyle
// @require        http://code.jquery.com/jquery-1.10.1.min.js
// @run-at         document-end
// ==/UserScript==
$(document).ready(function(){
    $('span[style*="font-size:10px;color: #888;"]').hide();
    
    var sContent = $('div[class*="content_spoiler"]');
    sContent.hide();
    var show_spoiler = '&nbsp&nbsp<a class="show-spoiler" class="btn btn-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px; border-radius:3px;" href="javascript:void(0);" title="Show Spoiler">Show</a>';
    var hide_spoiler = '<a class="hide-spoiler" class="btn bnt-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide Spoiler">Hide</a>';
    
    $('#bbcode_div > i').after(show_spoiler, hide_spoiler);
    $('.hide-spoiler').hide();
    
    $('.show-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).next().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.show();
    });
    
    $('.hide-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).prev().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.hide();
    });
});


mKse.v0.3.1.user.js
// ==UserScript==
// @name           mKaskus Spoiler Enabler
// @namespace      https://openuserjs.org/users/zackad/
// @homepageURL    http://www.kaskus.co.id/profile/4125324
// @description    Spoiler di m.kaskus layaknya versi desktop
// @author         zackad
// @version        0.3.1
// @include        http://m.kaskus.co.id/post/*
// @include        http://m.kaskus.co.id/thread/*
// @include        http://m.kaskus.co.id/lastpost/*
// @include        /^https?://www.kaskus.co.id/thread/*/
// @include        /^https?://www.kaskus.co.id/lastpost/*/
// @include        /^https?://www.kaskus.co.id/post/*/
// @include        /^https?://www.kaskus.co.id/group/discussion/*/
// @include        /^https?://www.kaskus.co.id/show_post/*/
// @license        MIT License
// @require        http://code.jquery.com/jquery-1.10.1.min.js
// @run-at         document-end
// ==/UserScript==
$(document).ready(function(){
    $('span[style*="font-size:10px;color: #888;"]').hide();
    
    if (window.location.href.indexOf("m.kaskus.co.id") > -1) {
    var sContent = $('div[class*="content_spoiler"]');
    sContent.hide();
    var show_all = '<a class="show-all" class="btn bnt-sm" style="margin: 0 10px;background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Show All Spoiler">Show all spoiler</a>';
    var hide_all = '<a class="hide-all" class="btn bnt-sm" style="margin: 0 10px;background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide All Spoiler">Hide all spoiler</a>';
    var show_spoiler = '&nbsp&nbsp<a class="show-spoiler" class="btn btn-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px; border-radius:3px;" href="javascript:void(0);" title="Show Spoiler">Show</a>';
    var hide_spoiler = '<a class="hide-spoiler" class="btn bnt-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide Spoiler">Hide</a>';
    
    $('#bbcode_div > i').after(show_spoiler, hide_spoiler, show_all, hide_all);
    $('.hide-spoiler').hide();
    $('.hide-all').hide();
    
    $('.show-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).next().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.show();
    });
    
    $('.hide-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).prev().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.hide();
    });
    
    $('.show-all').click(function(){
        sContent.show();
        $('.show-all').hide();
        $('.show-spoiler').hide();
        $('.hide-spoiler').show();
        $('.hide-all').show();
    });
    
    $('.hide-all').click(function(){
        sContent.hide();
        $('.hide-all').hide();
        $('.hide-spoiler').hide();
        $('.show-spoiler').show();
        $('.show-all').show();
    });
}
});


mKse.v0.1.user.dev.js
// ==UserScript==
// @name           mKaskus Spoiler Enabler
// @namespace      https://openuserjs.org/users/zackad/scripts
// @description    Spoiler di m.kaskus layaknya versi desktop
// @author         zackad
// @version        0.1
// @include        http://m.kaskus.co.id/post/*
// @include        http://m.kaskus.co.id/thread/*
// @include        http://m.kaskus.co.id/lastpost/*
// @license        MIT License
// @grant          GM_addStyle
// @require        http://code.jquery.com/jquery-1.10.1.min.js
// @run-at         document-end
// ==/UserScript==
$(document).ready(function(){
    var sContent = $('div[class*="content_spoiler"]');
    sContent.hide();

    var show_spoiler = '&nbsp&nbsp<a class="show-spoiler" class="btn btn-sm" style="background: #aaa;color: #FFF;padding:0 4px; font-size:10px; radius:2px;" href="javascript:void(0);" title="Show Spoiler">Show</a>';
    var hide_spoiler = '<a class="hide-spoiler" class="btn bnt-sm" style="background: #aaa;color: #fff;padding:0 4px; font-size:10px: radius:2px;" href="javascript:void(0);" title="Hide Spoiler">Hide</a>';
    
    $('#bbcode_div > i').append(show_spoiler, hide_spoiler);
    $('.hide-spoiler').hide();
    
    $('.show-spoiler').click(function(){
        var parent = $(this).parent().parent().parent();
        $(this).hide();
        $(this).next().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.show();
    });
    
    $('.hide-spoiler').click(function(){
        var parent = $(this).parent().parent().parent();
        $(this).hide();
        $(this).prev().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.hide();
    });
});


legacy/mKse.v0.1.user.js
// ==UserScript==
// @name           mKaskus Spoiler Enabler
// @namespace      https://openuserjs.org/users/zackad/scripts
// @description    Spoiler di m.kaskus layaknya versi desktop
// @author         zackad
// @version        0.1
// @include        http://m.kaskus.co.id/post/*
// @include        http://m.kaskus.co.id/thread/*
// @include        http://m.kaskus.co.id/lastpost/*
// @license        MIT License
// @grant          GM_addStyle
// @require        http://code.jquery.com/jquery-1.10.1.min.js
// @run-at         document-end
// ==/UserScript==
/*
	initial version
*/
$(document).ready(function(){
    var sContent = $('div[class*="content_spoiler"]');
    sContent.hide();

    var show_spoiler = '&nbsp&nbsp<a class="show-spoiler" class="btn btn-sm" style="background: #aaa;color: #FFF;padding:0 4px; font-size:10px; radius:2px;" href="javascript:void(0);" title="Show Spoiler">Show</a>';
    var hide_spoiler = '<a class="hide-spoiler" class="btn bnt-sm" style="background: #aaa;color: #fff;padding:0 4px; font-size:10px: radius:2px;" href="javascript:void(0);" title="Hide Spoiler">Hide</a>';
    
    $('#bbcode_div > i').append(show_spoiler, hide_spoiler);
    $('.hide-spoiler').hide();
    
    $('.show-spoiler').click(function(){
        var parent = $(this).parent().parent().parent();
        $(this).hide();
        $(this).next().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.show();
    });
    
    $('.hide-spoiler').click(function(){
        var parent = $(this).parent().parent().parent();
        $(this).hide();
        $(this).prev().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.hide();
    });
});


legacy/mKse.v0.3.2.user.js
// ==UserScript==
// @name           mKaskus Spoiler Enabler
// @namespace      https://openuserjs.org/users/zackad/
// @homepageURL    http://www.kaskus.co.id/profile/4125324
// @description    Spoiler di m.kaskus layaknya versi desktop
// @author         zackad
// @version        0.3.2
// @include        http://m.kaskus.co.id/post/*
// @include        http://m.kaskus.co.id/thread/*
// @include        http://m.kaskus.co.id/lastpost/*
// @include        /^https?://www.kaskus.co.id/thread/*/
// @include        /^https?://www.kaskus.co.id/lastpost/*/
// @include        /^https?://www.kaskus.co.id/post/*/
// @include        /^https?://www.kaskus.co.id/group/discussion/*/
// @include        /^https?://www.kaskus.co.id/show_post/*/
// @grant          GM_addStyle
// @license        MIT License
// @require        http://code.jquery.com/jquery-1.10.1.min.js
// @run-at         document-end
// ==/UserScript==
$(document).ready(function(){
    $('span[style*="font-size:10px;color: #888;"]').hide();
    
    if (window.location.href.indexOf("m.kaskus.co.id") > -1) {
    var sContent = $('div[class*="content_spoiler"]');
    sContent.hide();
    var show_all = '<a class="show-all" class="btn bnt-sm" style="margin: 0 10px;background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Show All Spoiler">Show all spoiler</a>';
    var hide_all = '<a class="hide-all" class="btn bnt-sm" style="margin: 0 10px;background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide All Spoiler">Hide all spoiler</a>';
    var show_spoiler = '&nbsp&nbsp<a class="show-spoiler" class="btn btn-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px; border-radius:3px;" href="javascript:void(0);" title="Show Spoiler">Show</a>';
    var hide_spoiler = '<a class="hide-spoiler" class="btn bnt-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide Spoiler">Hide</a>';
    
    $('#bbcode_div > i').after(show_spoiler, hide_spoiler, show_all, hide_all);
    $('.hide-spoiler').hide();
    $('.hide-all').hide();
    
    $('.show-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).next().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.show();
    });
    
    $('.hide-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).prev().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.hide();
    });
    
    $('.show-all').click(function(){
        sContent.show();
        $('.show-all').hide();
        $('.show-spoiler').hide();
        $('.hide-spoiler').show();
        $('.hide-all').show();
    });
    
    $('.hide-all').click(function(){
        sContent.hide();
        $('.hide-all').hide();
        $('.hide-spoiler').hide();
        $('.show-spoiler').show();
        $('.show-all').show();
    });
}
});


mKse.user.dev.js
// ==UserScript==
// @name           mKaskus Spoiler Enabler
// @namespace      https://openuserjs.org/users/zackad/
// @homepageURL    http://www.kaskus.co.id/profile/4125324
// @description    Spoiler di m.kaskus layaknya versi desktop
// @author         zackad
// @version        0.3.6.7
// @include        http://m.kaskus.co.id/*
// @include        /^https?://www.kaskus.co.id/thread/*/
// @include        /^https?://www.kaskus.co.id/lastpost/*/
// @include        /^https?://www.kaskus.co.id/post/*/
// @include        /^https?://www.kaskus.co.id/group/discussion/*/
// @include        /^https?://www.kaskus.co.id/show_post/*/
// @grant          GM_addStyle
// @license        MIT License
// @require        http://code.jquery.com/jquery-1.10.1.min.js
// @require        http://cdn.kaskus.com/themes_2.0/js/ajaxfileupload.js
// @run-at         document-end
// ==/UserScript==
/*
    LATEST UPDATE
    v0.3.6.7
    - use decodeURI instead replace
    - jQuery selector used
    - getTID function fixed
    v0.3.6.6
    - replace percent-encoding in link
    v0.3.6.5
    - redirect link bug fixed
    - experimental mode, upload file kaskus host
*/
$(document).ready(function(){
    /*===========================================
      GLOBAL SETTINGS
    *\===========================================*/
    var Settings = {
        SHOW_ORIGIN_LINK: false,   //ganti nilainya menjadi 'true' untuk menampilkan grey link
        SHOW_IMAGE_ONCLICK: true,  //false = buka image di tab baru, true = buka image langsung ditempat
        SHOW_IMAGE_SIZE: true,     //false = jangan tampilkan size gambar, true = tampilkan size gambar
        DECIMAL_POINT: 2,          //angka dibelakang koma size gambar
        SHOW_LASTPAGE: true,       //open thread's last page
        TO_WAP_LINK: true,         //convert kaskus link to wap version (m.kaskus)
        //experimental mode
        SHOW_WHO_POSTED: false,    //false = don't show, true = show , MASIH TAHAP DEVELOPMENT
        QUICK_LINKS: false         //show quick links
    };
    var QL = new Array(
        'Computer,19',
        'Anime,26',
        'ISN,183',
        'CCPB,14',
        'Linux,65'
    );
    //console.log(QL);
    /*===========================================
      UPLOAD FILE
    *\===========================================*/
/*
    var browse = '<input type="file" onchange="ajaxFileUpload();" name="forumimg" id="browse" class="browse" />';
    var ajaxScript = 'http://cdn.kaskus.com/themes_2.0/js/ajaxfileupload.js';
    var insertAjax = '<script type="text/javascript" src="'+ ajaxScript + '"></script><script type="text/javascript" src="http://code.jquery.com/jquery-1.10.1.min.js"></script>';
    $('#search').after(browse);
    $('body').append(insertAjax);
*/    
    /*===========================================
      REDIRECT LINK REMOVER [thanks : AMZZZMA]
    *\===========================================*/
    if (window.location.href.indexOf('m.kaskus.co.id') > -1) {
        var aEls = $('#content-wrapper .entry-content a');
    } else {
        var aEls = $('.postlist .entry a');
    }
    for (var i = 0, aEl; aEl = aEls[i]; i++) {
        //console.log(aEls.href);
        aEl.href = unescape(decodeURI(aEl.href));
        aEl.href = aEl.href.replace('http://m.kaskus.co.id/redirect?url=','');
        aEl.href = aEl.href.replace('http://www.kaskus.co.id/redirect?url=','');
    }
    
    if (Settings.TO_WAP_LINK == true && window.location.href.indexOf('m.kaskus.co.id') > -1) {
        var aEls = $('#content-wrapper .entry-content a');
        for (var i = 0, aEl; aEl = aEls[i]; i++) {
            aEl.href = aEl.href.replace('http://www.kaskus.co.id/','http://m.kaskus.co.id/');
            aEl.href = aEl.href.replace('http://kaskus.co.id/','http://m.kaskus.co.id/');
        }
    }
    /*===========================================
      SHOW LAST PAGE LINK
    *\===========================================*/
    if (Settings.SHOW_LASTPAGE == true && window.location.href == 'http://m.kaskus.co.id/myforum') {
        var list = $('#content-wrapper .post-list .list-entry');
        list.each(function(){
            var b = $(this).children('a').attr('href');
            //console.log(b);
            var TID = getTID(b);
            //console.log(TID);
            if (TID.length > 5) {
                var lastpage = 'http://m.kaskus.co.id/lastpost/' + TID;
                //console.log(TID);
                lastpage = '<span><a href="' + lastpage + '"> | Last Page | </a></span>';
                $(this).children('.sub-meta')
                    .append(lastpage);
            }
        });
    }
    
    
    /*===========================================
      SHOW QUICK LINKS
    *\===========================================*/
    if (Settings.QUICK_LINKS == true) {
        var qlWrapper = '<div id="ql"></div>';
        $('#search').after(qlWrapper);
        //QL.each(function(){
            
        //}
        
        $('ql').append(QL);
    }
    
    /*===========================================
      WHO POSTED ON THREAD  
    *\===========================================*/
    if (Settings.SHOW_WHO_POSTED == true) {
        var list = $('#content-wrapper .post-list .list-entry');
        
        list.each(function(){
            var tLink = $(this).children('a').attr('href');
            var TID = getTID(tLink);
            console.log(TID); //log thread ID
            if (TID.length > 5){
                var urlThread = "http://kaskus.co.id/misc/whoposted/"+ TID;
                
                //open whoposted on new tab :hammer:
                var who = $('<a>Who Posted</a>')
                    .attr('href', urlThread)
                    .attr('target', '_blank');
                var span = $('<span></span>').append(who);
                $(this).children('.sub-meta')
                    .append(span);
            }
            });
    }
    /*===========================================
      Origin Link Hider
    *\===========================================*/
    if (Settings.SHOW_ORIGIN_LINK == false) {
    $('span[style*="font-size:10px;color: #888;"]').hide();
    }
    
    /*===========================================
      Load Image on Click
    *\===========================================*/
    if (Settings.SHOW_IMAGE_ONCLICK == true) {
    var image = $('a[href$=".jpg"], a[href$=".png"], a[href$=".gif"], a[href$=".JPG"], a[href$=".PNG"], a[href$=".JPEG"], a[href$=".jpeg"], a[href$=".GIF"], a[href^="http://puu.sh/"]').css({'background-color': '#333', 'color' : 'white'});
    image.each(function(){
        var temp = $(this);
        temp.attr({data : $(this).attr('href'), alt : "Click to Load Image"}).attr('href', 'javascript:void(0);');
        
        // get image size in bytes
        if (Settings.SHOW_IMAGE_SIZE == true) {
        var xhr = $.ajax({
            type : 'HEAD',
            url : $(this).attr('data'),
            success : function(){
                var size = xhr.getResponseHeader('Content-Length');
                if (size > 1048576){
                    size = size/1048576;// + ' MB';
                    size = new String(size);
                    var charLoc = size.indexOf(".");// + Settings.DECIMAL_POINT;
                    if (charLoc > -1) {
                        charLoc = parseInt(charLoc) + parseInt(Settings.DECIMAL_POINT) + 1;
                        //console.log(charLoc);
                        size = size.substr(0,charLoc) + ' MB';
                    }
                } else if (size > 1024){
                    size = size/1024;// + ' KB';
                    size = new String(size);
                    var charLoc = size.indexOf(".");// + Settings.DECIMAL_POINT;
                    if (charLoc > -1) {
                        charLoc = parseInt(charLoc) + parseInt(Settings.DECIMAL_POINT) + 1;
                        //console.log(charLoc);
                        size = size.substr(0,charLoc) + ' KB';
                    }
                } else {
                    size = size + ' Bytes';
                }
                temp.after('<span style = "font-size : 10px; color : #00f;"> ' + size + '</span>');//.css({'color' : 'blue', 'font-size' : '8px'});
                //alert(size);
            }});
        }
    });
    image.click(function(){
       var link = $(this).attr('data');
        console.log(link);
       var img = $("<img />").attr('src', link);
       $(this).after(img);
       $(this).next().next().hide();
       $(this).hide();
    });
    }
    /*======================================
        Forum Activities Link
    \*======================================*/
        var quoted = $('#subnav span a').has('sup').last();//.text("Forum Activities");
        if (quoted.children('sup').text() > 0) {
            quoted.attr('href', '/myforum/myquotedpost');
        } else {
            quoted.attr('href', '/myforum');
        }
    
    /*===========================================================
        SPOILER ENABLER 
    *\===========================================================*/
    if (window.location.href.indexOf("m.kaskus.co.id") > -1) {
    var sContent = $('div[class*="content_spoiler"]');
    sContent.hide();
    var show_all = '<a class="show-all" class="btn bnt-sm" style="margin: 0 10px;background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Show All Spoiler Inside">Show all</a>';
    var hide_all = '<a class="hide-all" class="btn bnt-sm" style="margin: 0 10px;background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide All Spoiler Inside">Hide all</a>';
    var show_spoiler = '&nbsp&nbsp<a class="show-spoiler" class="btn btn-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px; border-radius:3px;" href="javascript:void(0);" title="Show Spoiler">Show</a>';
    var hide_spoiler = '<a class="hide-spoiler" class="btn bnt-sm" style="background: #ccc;color: #484848;text-decoration:none; padding:0 4px; font-size:10px: border-radius:3px;" href="javascript:void(0);" title="Hide Spoiler">Hide</a>';
    //var toggle_link = '<a class="toggle-link" class="btn btn-sm" style="background: #F7941D;color: #FFF; position:fixed; z-index:100; top:0 font-size:15px;" href="javascript:void(0);" title="Show Link">Show Origin Link</a>';
    $('#bbcode_div > i').after(show_spoiler, hide_spoiler, show_all, hide_all);
    $('.hide-spoiler').hide();
    $('.hide-all').hide();
    
    $('.show-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).next().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.show();
    });
    
    $('.hide-spoiler').click(function(){
        var parent = $(this).parent().parent();
        $(this).hide();
        $(this).prev().show();
        
        var currentSpoiler = $(parent).children('#bbcode_spoiler_content').children('#bbcode_inside_spoiler');
        currentSpoiler.hide();
    });
    
    $('.show-all').click(function(){
        sContent.show();  
        $('.show-all').hide();
        $('.show-spoiler').hide();
        $('.hide-spoiler').show();
        $('.hide-all').show();
    });
    
    $('.hide-all').click(function(){
        sContent.hide();
        $('.hide-all').hide();
        $('.hide-spoiler').hide();
        $('.show-spoiler').show();
        $('.show-all').show();
    });
    /*===========================================
      FUNCTION DECLALARION
    *\===========================================*/
    function toggle_spoiler(){
        sContent.toggle();
        $('.hide-all').toggle();
        $('.show-all').toggle();
        $('.hide-spoiler').toggle();
        $('.show-spoiler').toggle();
    }
    //get thread ID
    function getTID(a){
        var ID = new String(a);
        a = ID.split("/");
        var tID = a[2];
        return tID;
    }
    
    /* Hotkey */
    window.addEventListener('keydown', function(e) {
    var keyCode = e.keyCode;
    var CSA = [e.ctrlKey, e.shiftKey, e.altKey];
    //console.log(keyCode);
    //console.log(String(CSA) + '; '+keyCode);
    
    // caseof : Shift+X
    if( e.shiftKey && keyCode == 88 ){
        toggle_spoiler()
    }
}, true);
}});